tests/test_helpscout_interrupted_sync.py

```diff
@@ -0,0 +1,192 @@
+from base import HelpscoutBaseTest
+from dateutil import parser as parser
+
+
+class HelpscoutInterruptedSyncTest(HelpscoutBaseTest):
+    """
+    Test to verify that if a sync is interrupted, then the next sync will continue
+    from the bookmarks and currently syncing stream.
+    """
+
+    def name(self):
+        return "tap_helpscout_tests_using_shared_token_chaining"
+
+    def test_name(self):
+        LOGGER.info("Interrupted Sync test for tap-helpscout")
+
+    def test_run(self):
+        """
+        Scenario: A sync job is interrupted. The state is saved with `currently_syncing`.
+                  The next sync job kicks off and the tap picks back up on that
+                  `currently_syncing` stream.
+
+        Expected State Structure:
+            {
+                "currently_syncing": "stream_name",
+                "bookmarks": {
+                    "stream_1": "2010-10-10T10:10:10.100000",
+                    "stream_2": "2010-10-10T10:10:10.100000"
+                }
+            }
+
+        Test Cases:
+        - Verify an interrupted sync can resume based on the `currently_syncing` and stream level bookmark value.
+        - Verify only records with replication-key values greater than or equal to the
+          stream level bookmark are replicated on the resuming sync for the interrupted stream.
+        - Verify the yet-to-be-synced streams are replicated following the interrupted stream in the resuming sync.
+        """
+
+        self.start_date = self.get_properties()["start_date"]
+        start_date_timestamp = self.dt_to_ts(self.start_date)
+
+        conn_id = connections.ensure_connection(self, payload_hook=self.preserve_refresh_token)
+        expected_streams = self.expected_streams()
+
+        # Run check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Select only the expected streams tables
+        test_catalogs_all_fields = [catalog for catalog in found_catalogs
+                                    if catalog.get('tap_stream_id') in expected_streams]
+
+        # Catalog selection
+        self.perform_and_verify_table_and_field_selection(conn_id, test_catalogs_all_fields, select_all_fields=True)
+
+        # Run a sync job
+        self.run_and_verify_sync(conn_id)
+        first_sync_records = runner.get_records_from_target_output()
+
+        post_interrupted_sync_state = menagerie.get_state(conn_id)
+
+        # Simulated interrupted state to run 2nd sync
+        interrupted_sync_state = {
+            "currently_syncing": "customers",
+            "bookmarks": {
+                "conversations": "2021-12-02T16:47:26Z",
+                "customers": "2019-06-20T17:00:00Z"
+            }
+        }
+
+        # Set state for 2nd sync
+        menagerie.set_state(conn_id, interrupted_sync_state)
+
+        # Run sync after interruption
+        post_interrupted_sync_record_count_by_stream = self.run_and_verify_sync(conn_id)
```

Comment:
`post_interrupted_sync_record_count_by_stream` this variable is not getting used anywhere, hence, no need of assignment.

Reply:
Removed.